packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -367,276 +471,278 @@ private void createCaptureSessionWithSessionConfig(
-        null, (code, message) -> pictureCaptureRequest.error(code, message, null));
+    try {
+      captureSession.capture(previewRequestBuilder.build(), null, backgroundHandler);
+    } catch (CameraAccessException e) {
+      Log.i(TAG, "Error unlocking focus: " + e.getMessage());
+      dartMessenger.sendCameraErrorEvent(e.getMessage());
+      return;
+    }
   }
 
+  /** Cancel and reset auto focus state and refresh the preview session. */
   private void unlockAutoFocus() {
-    captureRequestBuilder.set(
-        CaptureRequest.CONTROL_AF_TRIGGER, CameraMetadata.CONTROL_AF_TRIGGER_CANCEL);
-    updateFocus(focusMode);
+    Log.i(TAG, "unlockAutoFocus");
     try {
-      cameraCaptureSession.capture(captureRequestBuilder.build(), null, null);
-    } catch (CameraAccessException ignored) {
+      if (captureSession == null) {
+        Log.i(TAG, "[unlockAutoFocus] captureSession null, returning");
+        return;
+      }
+      // Cancel existing AF state
+      previewRequestBuilder.set(
+          CaptureRequest.CONTROL_AF_TRIGGER, CameraMetadata.CONTROL_AF_TRIGGER_CANCEL);
+      captureSession.capture(previewRequestBuilder.build(), null, backgroundHandler);
+
+      // Set AF state to idle again
+      previewRequestBuilder.set(
+          CaptureRequest.CONTROL_AF_TRIGGER, CameraMetadata.CONTROL_AF_TRIGGER_IDLE);
+
+      captureSession.capture(previewRequestBuilder.build(), null, backgroundHandler);
+    } catch (CameraAccessException e) {
+      Log.i(TAG, "Error unlocking focus: " + e.getMessage());
+      dartMessenger.sendCameraErrorEvent(e.getMessage());
+      return;
     }
-    captureRequestBuilder.set(
-        CaptureRequest.CONTROL_AF_TRIGGER, CaptureRequest.CONTROL_AF_TRIGGER_IDLE);
 
     refreshPreviewCaptureSession(
         null,
-        (errorCode, errorMessage) -> pictureCaptureRequest.error(errorCode, errorMessage, null));
+        (errorCode, errorMessage) ->
+            dartMessenger.error(flutterResult, errorCode, errorMessage, null));
   }
 
   public void startVideoRecording(Result result) {
     final File outputDir = applicationContext.getCacheDir();
     try {
-      videoRecordingFile = File.createTempFile("REC", ".mp4", outputDir);
+      captureFile = File.createTempFile("REC", ".mp4", outputDir);
     } catch (IOException | SecurityException e) {
       result.error("cannotCreateFile", e.getMessage(), null);
       return;
     }
 
     try {
-      prepareMediaRecorder(videoRecordingFile.getAbsolutePath());
+      prepareMediaRecorder(captureFile.getAbsolutePath());
+
+      // Re-create autofocus feature so it's using video focus mode now
+      cameraFeatures.setAutoFocus(
+          cameraFeatureFactory.createAutoFocusFeature(cameraProperties, true));
       recordingVideo = true;
+
       createCaptureSession(
           CameraDevice.TEMPLATE_RECORD, () -> mediaRecorder.start(), mediaRecorder.getSurface());
       result.success(null);
     } catch (CameraAccessException | IOException e) {
```

Comment:
simplify the try/catch block.   it's ok to have multiple try/catch blocks, but it's generally not ideal to combine multiple failures into a single try/catch block.

Reply:
same below